### PR TITLE
fix(docs): use Demo component for ProgressBar

### PR DESCRIPTION
**Pull request by krosenk729**
fixes https://github.com/dequelabs/cauldron/issues/784

image

### CODE REVIEW

File: docs/patterns/components/ProgressBar/index.js [hunk 1]
```diff
@@ -42,79 +42,54 @@ const ProgressBarDemo = () => {
...
-          defaultProps={{}}
-        />
-      </div>
+      <Demo
+        component={ProgressBar}
+        componentDescription={
+          'Progress bars are used to indicate the status of some progress activity happening, such as number of items completed.'
+        }
+        states={[
+          { 'arial-label': 'Progress', progress: 75 },
```
Comment (anastasialanz) [inline docs/patterns/components/ProgressBar/index.js:53]:
```suggestion
          { 'aria-label': 'Progress', progress: 75 },
```

File: docs/patterns/components/ProgressBar/index.js [hunk 2]
```diff
@@ -42,79 +42,54 @@ const ProgressBarDemo = () => {
...
+      >
+        <h2>Animated</h2>
+        <p>
+          If the progress indicates that some activity is currently progressing,
+          the progress bar will update the status of the progress bar based on
+          the current <code>progress</code> prop:
+        </p>
+        <Code language="javascript" role="region" tabIndex={0}>
+          {`<ProgressBar aria-label="Progress" progress={progress} />`}
+        </Code>
```
Comment (anastasialanz) [inline docs/patterns/components/ProgressBar/index.js:87]: Can we move this code snippet to the bottom of the example similar to how the other ones are displayed above?

Screen Shot 2022-10-11 at 10 29 44 AM

**Review by anastasialanz — CHANGES_REQUESTED** [review]
@krosenk729 Looks pretty good! I have one small fix and one suggestion. Thanks again!

File: docs/patterns/components/ProgressBar/index.js (hunk 1 above)
Comment (krosenk729) [reply docs/patterns/components/ProgressBar/index.js]: ah thank you!

File: docs/patterns/components/ProgressBar/index.js (hunk 2 above)
Comment (krosenk729) [reply docs/patterns/components/ProgressBar/index.js]: done!

**Review by anastasialanz — APPROVED** [review]

**Comment by anastasialanz** [comment]
🚨 reviewed for security 🚨

**Pull request merged**